File: LABFINAL/src/UserMediator.java


import java.util.ArrayList;
import java.util.List;

public class UserMediator {
    private List<User> users;

    public UserMediator() {
        users = new ArrayList<>();
    }

    public void addUser(User user) {
        if (!users.contains(user)) {
            users.add(user);
        }
    }

    public void removeUser(User user) {
        users.remove(user);
    }

    public void notifyUsers(String message) {
        for (User user : users) {

        }

    } }

File: TaskManager/TaskManager/TaskManager/Mediator/UserMediator.cs
namespace TaskManager.Model;

public class UserMediator
{
    public List<User> _users;

    public UserMediator()
    {
        _users = new List<User>();
    }

    public void AddUser(User user)
    {
        if (_users.Contains(user))
        {
            return;
        }
        
        _users.Add(user);
    }

    public void RemoveUser(User user)
    {
        _users.Remove(user);
    }

    public void NotifyUsers(string message)
    {
        foreach (User user in _users)
        {
            user.ReceiveEmail(message);
        }
    }
}